src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3797,9 +3789,12 @@ void ride_construction_tooldown_construct(ScreenCoordsXY screenCoords)
         z = _trackPlaceZ;
     }
 
+    // search for z value to build at, up to max ride height
+    int numAttempts = (z <= (8 * MAX_TRACK_HEIGHT) ? MAX_TRACK_HEIGHT - (z / 8) + 1 : 2);
+
     if (ride->type == RIDE_TYPE_MAZE)
     {
-        for (int32_t zAttempts = 41; zAttempts >= 0; zAttempts--)
+        for (int32_t zAttempts = numAttempts; zAttempts > 0; zAttempts--)
```

Comment:
Is there any reason this is done backwards?

Reply:
I don't think so, maybe the original author thought that counting down is more intuitive for attempts?